garak/generators/huggingface.py

```diff
@@ -236,12 +297,13 @@ def _call_model(
     ) -> List[Union[str, None]]:
         """Take a conversation as a list of dictionaries and feed it to the model"""
 
+        self._load_client()
         # If conversation is provided as a list of dicts, create the conversation.
         # Otherwise, maintain state in Generator
         if isinstance(prompt, str):
             self.conversation.add_message({"role": "user", "content": prompt})
             self.conversation = self.generator(self.conversation)
-            generations = [self.conversation[-1]["content"]]
+            generations = [self.conversation[-1]["content"]]  # what is this doing?
```

Comment:
This class may become deprecated once #644 lands

Reply:
Yeah, this is a bit of a kluge to get conversations here, but this will pull the last message in the conversation from the generator. 
`self.conversation[-1]` will get the last element in the conversation ("role": "assistant" or "role": "bot" depending on templating) and pull the "content" element out of it. `self.conversation` is a `Conversation` object, an abstract class built on top of `dict`.

Reply:
The comment is missing context, it was focused on the line setting a new local `generations` value that is never consumed, I `think` it can simply be removed however since there is no clear test for this I just added a comment.